activerecord/test/cases/adapters/postgresql/partitions_test.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+
+class PostgreSQLPartitionsTest < ActiveRecord::PostgreSQLTestCase
```

Comment:
This test should be run only when this feature is supported. What about to add method to `PostgreSQLTestCase` checking if feature is available? You can get current version via `ActiveRecord::Base.connection.postgresql_version`.

```ruby
def postgresql_version     
  ActiveRecord::Base.connection.postgresql_version
end

def supports_partitioned_tables?
  postgresql_version >= 100000
end
```

in test:

```ruby
def test_partitions_table_exists
  skip unless supports_partitioned_tables?
  ...
end
```

This is similar how adapter supported features are detected, but it is probably not worth it to introduce this feature into adapter, since it is used only for tests now.

Also direct version check is possible as used in https://github.com/rails/rails/blob/4f0eb1ccd9652118b05d42f152adaa4df7dc3388/activerecord/test/cases/adapters/postgresql/transaction_test.rb#L97